src/lib/Scenes/Artwork/Components/CommercialButtons.tsx

```diff
@@ -0,0 +1,226 @@
+import { Button, Spacer } from "@artsy/palette"
+import { CommercialButtons_artwork } from "__generated__/CommercialButtons_artwork.graphql"
+import { CommercialButtonsOfferOrderMutation } from "__generated__/CommercialButtonsOfferOrderMutation.graphql"
+import { CommercialButtonsOrderMutation } from "__generated__/CommercialButtonsOrderMutation.graphql"
+import SwitchBoard from "lib/NativeModules/SwitchBoard"
+import React from "react"
+import { commitMutation, createFragmentContainer, graphql, RelayProp } from "react-relay"
+
+interface Props {
+  artwork: CommercialButtons_artwork
+  relay: RelayProp
+  editionSetID?: string
+}
+
+export interface State {
+  isCommittingCreateOrderMutation: boolean
+  isCommittingCreateOfferOrderMutation: boolean
+}
+
+export class CommercialButtons extends React.Component<Props, State> {
+  state = {
+    isCommittingCreateOfferOrderMutation: false,
+    isCommittingCreateOrderMutation: false,
+  }
+
+  onMutationError(error) {
+    // FIXME: Handle error
+    console.log("src/lib/Scenes/Artwork/Components/CommercialButtons.tsx", error)
+  }
+
+  handleCreateOrder() {
+    const { relay, artwork, editionSetID } = this.props
+    const { isCommittingCreateOrderMutation, isCommittingCreateOfferOrderMutation } = this.state
+    const { internalID } = artwork
+    if (isCommittingCreateOrderMutation || isCommittingCreateOfferOrderMutation) {
+      return
+    }
+    this.setState({ isCommittingCreateOrderMutation: true }, () => {
+      if (relay && relay.environment) {
+        commitMutation<CommercialButtonsOrderMutation>(relay.environment, {
+          mutation: graphql`
+            mutation CommercialButtonsOrderMutation($input: CommerceCreateOrderWithArtworkInput!) {
+              commerceCreateOrderWithArtwork(input: $input) {
+                orderOrError {
+                  ... on CommerceOrderWithMutationSuccess {
+                    __typename
```

Comment:
Should this go on the `orderOrError`, rather than within the `CommerceOrderWithMutationSuccess` spread?

Reply:
I'm not sure about the answer for this, is there someone else with more context we could reach out to?

Reply:
Thinking about this again, I'm fairly certain it should go into `orderOrError`, since `__typename` exists on both, and not just `CommerceOrderWithMutationSuccess`. Here's another use of it:

https://github.com/artsy/emission/blob/6567bdce2e708520f6f02a8a837fe8e8b87052cb/src/lib/Components/Inbox/Conversations/ConversationSnippet.tsx#L188-L190